packages/@sanity/components/src/fileinput/DropZone.js

```diff
@@ -1,43 +1,143 @@
 import React, {PropTypes} from 'react'
 import DropZone from 'react-dropzone'
+import {get, includes} from 'lodash'
 
 import styles from 'part:@sanity/components/fileinput/dropzone-style'
 
 export default class Dropzone extends React.PureComponent {
   static propTypes = {
     className: PropTypes.string,
-    multiple: PropTypes.bool
+    multiple: PropTypes.bool,
+    icon: PropTypes.func,
+    ghost: PropTypes.bool,
+    accept: PropTypes.string
   }
 
   static defaultProps = {
     multiple: true
   }
 
+  state = {
+    hasUnAcceptedFiles: false,
```

Comment:
Later in the code I check if there is files that is not accepted. Setting error state based on acceptsAll=false is confusing.